knack/cli.py

```diff
@@ -236,7 +236,7 @@ def invoke(self, args, initial_invocation_data=None, out_file=None):
                 if cmd_result and cmd_result.result is not None:
                     formatter = self.output.get_formatter(output_type)
                     self.output.out(cmd_result, formatter=formatter, out_file=out_file)
-                self.raise_event(EVENT_CLI_SUCCESSFUL_EXECUTE, result=cmd_result.result)
+                self.raise_event(EVENT_CLI_SUCCESSFUL_EXECUTE, result=cmd_result)
```

Comment:
Send the full `CommandResultItem` in case post-output hint relies on other attributes:

https://github.com/microsoft/knack/blob/587abe594aa5bf1fcebd3e373f5304f285f0eb0f/knack/util.py#L23-L32

Reply:
Can we add another parameter into `EVENT_CLI_SUCCESSFUL_EXECUTE` to keep `cmd_result`? Although `azure-cli` didn't use `EVENT_CLI_SUCCESSFUL_EXECUTE` event now, but I'm afraid some other cli depend on knack might face breaking change in `result` type.

Reply:
Adding arg in function invocation may also trigger breaking change if the callee doesn't accept the new arg.

We never guaranteed there will be no breaking changes:

https://github.com/microsoft/knack

> The project is in initial development phase. We recommend pinning to at least a specific minor version when marking knack as a dependency in your project.